### PR TITLE
43565 user register

### PR DESCRIPTION
**Pull request by yasin-buet**

### CODE REVIEW

File: app/Http/routes.php [hunk 1]
```diff
@@ -11,6 +11,22 @@
 |
 */
 
+// Route::get('/', function () {
```
Comment (KhanhLD) [inline app/Http/routes.php]: Please try to avoid adding unnecessary lines.
For examples: these lines from 14 -> 19. and line 25 in app/User.php

Comment (yasin-buet) [inline app/Http/routes.php]: ok,sure :)

File: app/Http/routes.php [hunk 2]
```diff
@@ -11,6 +11,30 @@
...
+// });
+
+
+
 Route::get('/', function () {
     return view('welcome');
 });
+
+
+<<<<<<< HEAD
```
Comment (wataridori) [inline app/Http/routes.php]: This file has conflict content. Please fix it.

File: app/User.php [hunk 3]
```diff
@@ -22,6 +22,11 @@ class User extends Model implements AuthenticatableContract,
      * @var string
      */
     protected $table = 'users';
+<<<<<<< HEAD
```
Comment (wataridori) [inline app/User.php]: This file was conflicted too

Comment (yasin-buet) [inline app/User.php]: conflict taken care of and rebased.

File: app/Http/routes.php (hunk 2 above)
Comment (yasin-buet) [inline app/Http/routes.php]: conflict removed and rebased.

File: app/Http/routes.php [hunk 4]
```diff
@@ -11,6 +11,20 @@
...
+
+
+// Authentication routes...
+Route::get('auth/login', 'Auth\AuthController@getLogin');
+Route::post('auth/login', 'Auth\AuthController@postLogin');
+Route::get('auth/logout', 'Auth\AuthController@getLogout');
+
+// Registration routes...
+Route::get('auth/register', 'Auth\AuthController@getRegister');
+Route::post('auth/register', 'Auth\AuthController@postRegister');
```
Comment (wataridori) [inline app/Http/routes.php]: You should use `Route::group()` to group these similar routes 
```php 
Route::group(['namespace' => 'Auth', 'prefix' => 'auth'], function() {
    Route::get('login', 'AuthController@getLogin');
    Route::post('login', 'AuthController@postLogin');
    Route::get('logout', 'AuthController@getLogout');
})
```

Comment (yasin-buet) [inline app/Http/routes.php]: done it.

File: app/Http/routes.php [hunk 5]
```diff
@@ -11,6 +11,19 @@
...
     return view('welcome');
 });
+
+
+Route::group(['namespace' => 'Auth', 'prefix' => 'auth'], function() {
+    Route::get('login', 'AuthController@getLogin');
+    Route::post('login', 'AuthController@postLogin');
+    Route::get('logout', 'AuthController@getLogout');
+	
+	Route::get('register', 'AuthController@getRegister');
```
Comment (wataridori) [inline app/Http/routes.php]: You should fix this line's indent.
Please replace `tab` with `4 spaces`

Comment (yasin-buet) [inline app/Http/routes.php]: corrected indentation.

**Comment by yasin-buet** [comment]
Today i am working creating a new branch .
My previous branch is not merged yet.
thanks

File: app/Http/routes.php (hunk 5 above)
Comment (yasin-buet) [inline app/Http/routes.php]: Today i am working creating a new branch .
My previous branch is not merged yet.
thanks

**Comment by wataridori** [comment]
OK

File: app/Http/routes.php [hunk 6]
```diff
@@ -11,6 +11,18 @@
 |
 */
 
+
```
Comment (KhanhLD) [inline app/Http/routes.php]: please don't added empty lines unnecessarily.

File: resources/views/auth/login.blade.php [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<form method="POST" action="/auth/login">
+    {!! csrf_field() !!}
```
Comment (KhanhLD) [inline resources/views/auth/login.blade.php]: http://laravelcollective.com/docs/5.1/html#csrf-protection

Comment (yasin-buet) [inline resources/views/auth/login.blade.php]: csrf protection added.

Comment (KhanhLD) [inline resources/views/auth/login.blade.php]: From the link I've posted above:
If you use the Form::open method with POST, PUT or DELETE the CSRF token will be added to your forms as a hidden field automatically.

What I mean is you should use Form::open so you don't have to add csrf_field() manually.

Please read the link above for more details and research more about it.

Comment (yasin-buet) [inline resources/views/auth/login.blade.php]: i am on it .

Comment (yasin-buet) [inline resources/views/auth/login.blade.php]: studied on forms and made changes.

File: app/Http/routes.php [hunk 8]
```diff
@@ -10,7 +10,18 @@
 | and give it the controller to call when that URI is requested.
 |
 */
-
 Route::get('/', function () {
     return view('welcome');
 });
+
+Route::group(['namespace' => 'Auth', 'prefix' => 'auth','as' => 'auth'], function() {
+    //Route::get ()
```
Comment (KhanhLD) [inline app/Http/routes.php]: delete

File: composer.json [hunk 9]
```diff
@@ -5,8 +5,10 @@
     "license": "MIT",
     "type": "project",
     "require": {
+        "laravelcollective/html": "5.1.*",
         "php": ">=5.5.9",
         "laravel/framework": "5.1.*"
+
```
Comment (KhanhLD) [inline composer.json]: delete

File: config/app.php [hunk 10]
```diff
@@ -145,6 +145,9 @@
         App\Providers\EventServiceProvider::class,
         App\Providers\RouteServiceProvider::class,
 
+        // laravel collective
```
Comment (KhanhLD) [inline config/app.php]: delete empty lines and unnecessary comment lines

File: config/app.php [hunk 11]
```diff
@@ -193,6 +196,9 @@
         'URL'       => Illuminate\Support\Facades\URL::class,
         'Validator' => Illuminate\Support\Facades\Validator::class,
         'View'      => Illuminate\Support\Facades\View::class,
+        // include for laravel collective0
```
Comment (KhanhLD) [inline config/app.php]: delete

File: resources/views/auth/login.blade.php [hunk 12]
```diff
@@ -0,0 +1,21 @@
+<!-- resources/views/auth/login.blade.php -->
+
+{!! Form::open(array('url' => 'auth/login')) !!}
```
Comment (KhanhLD) [inline resources/views/auth/login.blade.php]: use [] instead of array()

File: resources/views/auth/login.blade.php [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+    <div>
+        Email
+        <input type="email" name="email" value="{{ old('email') }}">
```
Comment (KhanhLD) [inline resources/views/auth/login.blade.php]: use Form helper for all inputs if possible

File: resources/views/auth/register.blade.php [hunk 14]
```diff
@@ -0,0 +1,32 @@
+<!-- resources/views/auth/register.blade.php -->
+
+{!! Form::open(array('url' => 'auth/register')) !!}
```
Comment (KhanhLD) [inline resources/views/auth/register.blade.php]: use [] instead of array()

File: resources/views/auth/register.blade.php [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+        {!! Form::email('email') !!}
+    </div>
+
+    <div>
+        {!! Form::label('password','Password',['class' => 'awsome']) !!}
+        {!! Form::password('password')!!}
+    </div>
+
+    <div>
+        {!! Form::label('password','Confirm Password',['class' => 'awsome']) !!}
```
Comment (KhanhLD) [inline resources/views/auth/register.blade.php]: password_confirmation

Comment (yasin-buet) [inline resources/views/auth/register.blade.php]: done.

Comment (KhanhLD) [inline resources/views/auth/register.blade.php]: why don't you change the label for password_confirmation ?

Comment (yasin-buet) [inline resources/views/auth/register.blade.php]: done.

**Comment by KhanhLD** [comment]
ok

**Pull request merged**